diff --git a/node_modules/next-auth/lib/index.js b/node_modules/next-auth/lib/index.js
index 2da51cc..adaaad5 100644
--- a/node_modules/next-auth/lib/index.js
+++ b/node_modules/next-auth/lib/index.js
@@ -37,10 +37,10 @@ function isReqWrapper(arg) {
 export function initAuth(config, onLazyLoad // To set the default env vars
 ) {
     if (typeof config === "function") {
-        return (...args) => {
+        return async (...args) => {
             if (!args.length) {
                 // React Server Components
-                const _headers = headers();
+                const _headers = await headers();
                 const _config = config(undefined); // Review: Should we pass headers() here instead?
                 onLazyLoad?.(_config);
                 return getSession(_headers, _config).then((r) => r.json());
@@ -82,10 +82,10 @@ export function initAuth(config, onLazyLoad // To set the default env vars
             });
         };
     }
-    return (...args) => {
+    return async (...args) => {
         if (!args.length) {
             // React Server Components
-            return getSession(headers(), config).then((r) => r.json());
+            return getSession(await headers(), config).then((r) => r.json());
         }
         if (args[0] instanceof Request) {
             // middleware.ts inline
